core/src/main/java/dev/morphia/mapping/codec/MorphiaMapPropertyCodecProvider.java

```diff
@@ -21,7 +21,7 @@
 import static dev.morphia.aggregation.codecs.ExpressionHelper.document;
 
 @SuppressWarnings("unchecked")
-class MorphiaMapPropertyCodecProvider extends MorphiaPropertyCodecProvider {
+public class MorphiaMapPropertyCodecProvider extends MorphiaPropertyCodecProvider {
```

Comment:
This doesn't need to be made public. your test class need only be a `PropertyCodecProvider`.

Reply:
reverted